invenio_preservation_sync/receivers.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Preservation-Sync is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Receiver for managing Preservation Sync events integration."""
+
+from flask import current_app, g
+from invenio_webhooks.models import Receiver
+from werkzeug.exceptions import BadRequest
+
+from invenio_preservation_sync.api import PreservationInfoAPI
+from invenio_preservation_sync.utils import get_mandatory_config
+
+from .errors import (
+    InvalidSenderError,
+    MissingConfigError,
+    PreservationAlreadyReceivedError,
+)
+
+
+class PreservationSyncReceiver(Receiver):
+    """Handle incoming notification from GitHub on a new release."""
+
+    def run(self, event):
+        """Process an event."""
+        try:
+            perm_policy = get_mandatory_config(
+                current_app, "PRESERVATION_SYNC_PERMISSION_POLICY"
+            )
+            pid_resolver = get_mandatory_config(
+                current_app, "PRESERVATION_SYNC_PID_RESOLVER"
+            )
```

Comment:
This feels like business layer, therefore I would create a layer for this. 

E.g. 

```python
# receivers.py
def run(self, event):
    # process the event and call the business layer (service)
    service.preserve(pid_id, revision_id, status)
```

```python
# service.py
def preserve(self):
    # check permissions
    ...
    # archive the record or not
```